web/src/utilities/utilities.js

```diff
@@ -636,6 +636,37 @@ export const Utilities = {
     }
   },
 
+  isClusterUpgrading(state) {
+    const normalizedState = this.clusterState(state);
+    return (
+      normalizedState === "Upgrading" || normalizedState === "Upgrading addons"
+    );
+  },
+
+  showClusterUpgradeModal(apps) {
```

Comment:
this name indicates an action will be taken, how about:
```suggestion
  shouldShowClusterUpgradeModal(apps) {
```